spring-cloud-data-rest-client/src/main/java/org/springframework/cloud/data/rest/client/TaskTemplate.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.rest.client;
+
+import java.util.Collections;
+import java.util.Map;
+
+import org.springframework.cloud.data.rest.resource.TaskDefinitionResource;
+import org.springframework.cloud.data.rest.util.DeploymentPropertiesUtils;
+import org.springframework.hateoas.UriTemplate;
+import org.springframework.util.Assert;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Implementation for {@link org.springframework.cloud.data.rest.client.TaskOperations}.
+ *
+ * @author Glenn Renfro
+ */
+public class TaskTemplate implements TaskOperations {
+
+	private static final String DEFINITIONS_PATH = "tasks/definitions";
+
+	private static final String DEPLOYMENTS_PATH = "tasks/deployments";
+
+	private static final String DEFAULT_PATH = "tasks/";
```

Comment:
How is that "default"? 

The uri template /tasks/{name} should not be used. Should be /tasks/definitions/{name} (to be changed at controller side as well)

Moreover, those things are not "paths", they are "relations".
The things that are looked up in the map could be named paths, though.